WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Onboarding Errors/InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import Yosemite
+
+class InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel: ObservableObject {
+    let completion: () -> ()
+    private let stores: StoresManager
+    private let noticePresenter: NoticePresenter
+
+    @Published var awaitingResponse = false
+
+    private var siteID: Int64? {
+        stores.sessionManager.defaultStoreID
+    }
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter,
+         completion: @escaping () -> ()) {
+        self.stores = stores
+        self.noticePresenter = noticePresenter
+        self.completion = completion
+    }
+
+    func skipTapped() {
+        guard let siteID = siteID else {
+            return completion()
+        }
+
+        let action = AppSettingsAction.setSkippedCashOnDeliveryOnboardingStep(siteID: siteID)
+        stores.dispatch(action)
+        completion()
+    }
+
+    func enableTapped() {
+        guard let siteID = siteID else {
+            return completion()
+        }
+
+        awaitingResponse = true
+
+        let action = PaymentGatewayAction.updatePaymentGateway(defaultCashOnDeliveryGateway(siteID: siteID)) { [weak self] result in
+            guard let self = self else { return }
+            guard result.isSuccess else {
+                DDLogError("💰 Could not update Payment Gateway: \(String(describing: result.failure))")
+                self.awaitingResponse = false
+                self.displayEnableCashOnDeliveryFailureNotice()
+                return
+            }
+
+            self.completion()
+        }
+        stores.dispatch(action)
+    }
```

Comment:
I'm having some trouble to understand this bit. From what I understand, the idea is to update the payment gateway by dispatching the action to store, for this, we must wait for the return value of the request to see if is successful or not. My confusion comes from the 2 `self` in the function:

I understand that we have 2 different `self` here, one for `Result<PaymentGateway, Error>` and another for `InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel`, which refers to the class instance itself.  My confusion comes when,  if I check the declaration of `[weak self]` this points me to another `InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel` which is `Optional` 🤔

Reply:
We'll go over this in pairing tomorrow, I think it'll be easier in person.